noxfiles/docker_nox.py

```diff
@@ -184,112 +184,28 @@ def build(session: nox.Session, image: str, machine_type: str = "") -> None:
     session.run(*build_command, external=True)
 
 
-def push_prod(session: nox.Session) -> None:
-    """
-    Contains the logic for pushing the suite of 'prod' images.
-
-    Pushed Image Examples:
-      - ethyca/fides:2.0.0
-      - ethyca/fides:latest
-
-      - ethyca/fides-privacy-center:2.0.0
-      - ethyca/fides-privacy-center:latest
-
-      - ethyca/fides-sample-app:2.0.0
-      - ethyca/fides-sample-app:latest
-    """
-    fides_image_name = get_current_image()
-    privacy_center_image_name = f"{PRIVACY_CENTER_IMAGE}:{get_current_tag()}"
-    sample_app_image_name = f"{SAMPLE_APP_IMAGE}:{get_current_tag()}"
-
-    privacy_center_latest = f"{PRIVACY_CENTER_IMAGE}:latest"
-    sample_app_latest = f"{SAMPLE_APP_IMAGE}:latest"
-
-    fides_buildx_command = generate_multiplatform_buildx_command(
-        [fides_image_name, IMAGE_LATEST], docker_build_target="prod"
-    )
-    session.run(*fides_buildx_command, external=True)
-
-    privacy_center_buildx_command = generate_multiplatform_buildx_command(
-        [privacy_center_image_name, privacy_center_latest],
-        docker_build_target="prod_pc",
-    )
-    session.run(*privacy_center_buildx_command, external=True)
-
-    sample_app_buildx_command = generate_multiplatform_buildx_command(
-        [sample_app_image_name, sample_app_latest],
-        docker_build_target="prod",
-        dockerfile_path="clients/sample-app",
-    )
-    session.run(*sample_app_buildx_command, external=True)
-
-
-def push_dev(session: nox.Session) -> None:
-    """
-    Push the bleeding-edge `dev` images.
-
-    Pushed Image Examples:
-      - ethyca/fides:dev
-      - ethyca/fides-privacy-center:dev
-      - ethyca/fides-sample-app:dev
-    """
-    privacy_center_dev = f"{PRIVACY_CENTER_IMAGE}:dev"
-    sample_app_dev = f"{SAMPLE_APP_IMAGE}:dev"
+def push_fides_images(session: nox.Session, tag_suffixes: List[str]):
```

Comment:
we had shared logic on our different "push" functions, so i tried to distill it down here. everything should work exactly as it had, just DRYing it up. and of course adding the ability to push `rc` and `prerelease` tags.

i'm thinking we should get some test coverage around the generated push commands though, just to be safe. i think it will require some mocking though, so it'll take a little bit of thought. i'll take a stab at that in a bit.

Reply:
true, we could do that, but i think it actually shouldn't be too bad if i just mock `session.run`...giving that a try now but may pivot to what you suggest.

Reply:
i got the mocking working pretty well here https://github.com/ethyca/fides/pull/3474/commits/b7cc405107f708a58543ce8e9697c67533bc030e so i'm gonna move forward with that, i think it's pretty easy to follow! let me know if you think otherwise and we can always adjust 👍